packages/teleport-repl-component/src/components/editor/editor.tsx

```diff
@@ -16,13 +16,34 @@ export class Editor {
   @Prop() public dark: boolean = false
   @State() public selectedCSSFlavour: string
   @State() public selectedFramework: 'vue' | 'react' = 'react'
+  @State() public showCssFile: boolean = false
   private jsonEditor: any
   private javascriptEditor: any
+  private cssEditor: any
 
   public componentDidLoad() {
     this.setupEditor()
   }
 
+  public handleTabClick() {
+    const el = document.getElementById('css-tab')
+    const editor = document.getElementById('cssEditor')
+    this.showCssFile = !this.showCssFile
+    if (this.showCssFile) {
+      this.openTab()
+    } else {
+      editor.style.display = 'none'
```

Comment:
shouldn't these styles be set based on `showCssFile` inside the `render` function?

Reply:
No, to get the render toggle functionality. On click of the header, I am listening to this func and setting the styles. I think stencil doesn't support for inline styles with state currently, so need to do using plain JS.

Reply:
the render function should be triggered any time you are changing something that is annotated with @State. so styles could be set based on that as well I think. have you tried it?

Reply:
Yes, inline styling like with state or prop is currently not supported, they support only class as of now

Reply:
then we could do class toggle, if state is X, add that class to the element

Reply:
Yeah makes sense, i will refactor by setting classes